controllers/ebpf/agent_controller.go

```diff
@@ -468,6 +471,32 @@ func (c *AgentController) setEnvConfig(coll *flowslatest.FlowCollector) []corev1
 			Name:  envMetricPrefix,
 			Value: "netobserv_agent_",
 		})
+		var promTLS *flowslatest.CertificateReference
+		switch coll.Spec.Agent.EBPF.Metrics.Server.TLS.Type {
+		case flowslatest.ServerTLSProvided:
+			promTLS = coll.Spec.Agent.EBPF.Metrics.Server.TLS.Provided
+			if promTLS == nil {
+				rlog.Info("EBPF agent metric tls configuration set to provided but none is provided")
+			}
+		case flowslatest.ServerTLSAuto:
+			promTLS = &flowslatest.CertificateReference{
+				Type:     "secret",
+				Name:     constants.EBPFAgentMetricsSvcName,
+				CertFile: "tls.crt",
+				CertKey:  "tls.key",
+			}
+		case flowslatest.ServerTLSDisabled:
+			// nothing to do there
+		}
+		if promTLS != nil {
```

Comment:
If I'm correct, it's missing the volume mount for the certificate?
We have a helper to do that:
```
		c.volumes.AddCertificate(promTLS, "prom-certs")
```

see for instance on FLP: https://github.com/netobserv/network-observability-operator/blob/main/controllers/flp/flp_common_objects.go#L295